ui/apps/platform/src/hooks/usePermissions.ts

```diff
@@ -1,32 +1,30 @@
-import { useSelector } from 'react-redux';
-import { createStructuredSelector } from 'reselect';
+import { createContext, useContext } from 'react';
 
-import { selectors } from 'reducers';
-import type { Access } from 'types/role.proto';
 import type { ResourceName } from 'types/roleResources';
 import { replacedResourceMapping } from 'constants/accessControl';
+import { Access } from 'types/role.proto';
 
 export type HasNoAccess = (resourceName: ResourceName) => boolean;
 export type HasReadAccess = (resourceName: ResourceName) => boolean;
 export type HasReadWriteAccess = (resourceName: ResourceName) => boolean;
 
+export const UserPermissionContext = createContext<{
+    userRolePermissions: { resourceToAccess: Partial<Record<ResourceName, Access>> };
+    isLoadingPermissions: boolean;
+}>({
+    userRolePermissions: { resourceToAccess: {} },
+    isLoadingPermissions: false,
+});
+
 type UsePermissionsResponse = {
     hasNoAccess: HasNoAccess;
     hasReadAccess: HasReadAccess;
     hasReadWriteAccess: HasReadWriteAccess;
     isLoadingPermissions: boolean;
 };
 
-const stateSelector = createStructuredSelector<{
-    userRolePermissions: { resourceToAccess: Record<ResourceName, Access> };
-    isLoadingPermissions: boolean;
-}>({
-    userRolePermissions: selectors.getUserRolePermissions,
-    isLoadingPermissions: selectors.getIsLoadingUserRolePermissions,
-});
-
 const usePermissions = (): UsePermissionsResponse => {
-    const { userRolePermissions, isLoadingPermissions } = useSelector(stateSelector);
+    const { userRolePermissions, isLoadingPermissions } = useContext(UserPermissionContext);
```

Comment:
In `useFeatureFlags`, you added error handling for when the hook is used outside the provider context. Should we add something like that here just to be safe?